### PR TITLE
[spryker-internal-jira] FAN-1: Creation of affiliate products

### PR DESCRIPTION
**Pull request by muhammad-kamran-spryker**
https://spryker.atlassian.net/projects/FAN/issues/FAN-1

### CODE REVIEW

File: src/Pyz/Yves/Product/Controller/ProductController.php [hunk 1]
```diff
@@ -22,10 +25,28 @@ class ProductController extends AbstractController
...
      *
-     * @return array
+     * @return array | RedirectResponse
+     * @throws \Exception
      */
     public function detailAction(StorageProductTransfer $storageProductTransfer)
     {
+        if ($storageProductTransfer->getIsAffiliateProduct()) {
+
+            $affiliateProductAccessLogTransfer = $this->getFactory()->createAffiliateProductAccessLogTransfer();
```
Comment (hhebbo) [inline src/Pyz/Yves/Product/Controller/ProductController.php]: there is no need for factory here, you can directly instantiate transfer objects in the code

File: src/Pyz/Yves/Product/ProductFactory.php [hunk 2]
```diff
@@ -111,4 +112,12 @@ public function createStorageProductMapperPlugin()
     {
         return new StorageProductMapperPlugin();
     }
+
+    /**
+     * @return AffiliateProductAccessLogTransfer
+     */
+    public function createAffiliateProductAccessLogTransfer()
```
Comment (hhebbo) [inline src/Pyz/Yves/Product/ProductFactory.php]: no need for this method

Comment (muhammad-kamran-spryker) [reply src/Pyz/Yves/Product/ProductFactory.php]: i thought it was a good idea to not use the `new` keyword in the controller :) should I still change it you think? thanks.

Comment (hhebbo) [reply src/Pyz/Yves/Product/ProductFactory.php]: yes, for transfer objects you can use new anywhere

Comment (muhammad-kamran-spryker) [reply src/Pyz/Yves/Product/ProductFactory.php]: done :+1:  thanks for the review

**Pull request merged**